lib/feature/browser/browser_tab_view/browser_tab_view.dart

```diff
@@ -29,14 +49,47 @@ class _BrowserTabViewState extends State<BrowserTabView> {
   // How long to wait before considering scroll event as not overscroll
   static const Duration _scrollTimerDelay = Duration(milliseconds: 100);
 
+  InAppWebViewController? _webViewController;
+
+  final _initialOptions = InAppWebViewGroupOptions(
+    ios: IOSInAppWebViewOptions(),
+    android: AndroidInAppWebViewOptions(),
+    crossPlatform: InAppWebViewOptions(),
+  );
+
+  @override
+  void didUpdateWidget(BrowserTabView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    onTabChanged(widget.tab, oldWidget.tab);
+  }
+
+  Future<void> onTabChanged(BrowserTab newTab, BrowserTab oldTab) async {
+    final url = await _webViewController?.getUrl();
+
+    if (newTab.url != oldTab.url && url != newTab.url) {
+      await _webViewController?.loadUrl(
+        urlRequest: URLRequest(
+          url: widget.tab.url,
+        ),
+      );
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return InAppWebView(
```

Comment:
Don't forget to look through logic from old app, because there can be a lot of different logic

Reply:
You can resolve it urself

Reply:
Yeah, but this is just a initial PR, it contains only tab switching logic.